packages/core/lib/command-utils.js

```diff
@@ -233,11 +233,11 @@ const deriveConfigEnvironment = function (detectedConfig, network, url) {
 
   if (configDefinesProvider) {
     // Use "provider" specified in the config to connect to the network
-    // along with the other specified network properties
+    // along with the other network properties
     configuredNetwork = {
-      ...detectedConfig.networks[network],
       network_id: "*",
-      provider: detectedConfig.networks[network].provider
+      provider: detectedConfig.networks[network].provider,
```

Comment:
Is this necessary if the entire network object, which should include the provider property, is spread out?

Reply:
Yeah! This is unnecessary as the network object will include the `provider` specified. I will remove this line. Thanks @cds-amal.